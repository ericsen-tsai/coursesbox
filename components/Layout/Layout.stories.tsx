import { ComponentStory, ComponentMeta } from '@storybook/react'

import Layout from './index'

export default {
  title: 'Content/Layout',
  component: Layout,
} as ComponentMeta<typeof Layout>

const Template: ComponentStory<typeof Layout> = (args) => <Layout {...args} />

export const BasicLayout = Template.bind({})
BasicLayout.args = {
  children: 'test text',
}
