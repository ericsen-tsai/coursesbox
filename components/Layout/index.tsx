import { ReactNode } from 'react'
import Link from 'next/link'
import styled from '@emotion/styled'

import StyledLink from '../StyledLink'
import Logo from '../Logo'
import Input from '../Input'
import IconButton from '../IconButton'

const Wrapper = styled.div`
  display: grid;
  gap: 0.1rem;
  color: ${({ theme }) => theme.font.regular};
  background-color: ${({ theme }) => theme.background};
  padding: 0.5rem;

  grid-template-areas:
    'header nav'
    'search search'
    'content content'
    'footer footer';

  nav {
    flex-direction: row;
    justify-content: flex-end;
    gap: 5vmin;
  }
  @media (min-width: 500px) {
    grid-template-columns: 1fr 3fr;
    grid-template-areas:
      'header nav'
      'search search'
      'content content'
      'footer footer';

    nav {
      flex-direction: row;
      justify-content: flex-end;
    }
  }

  @media (min-width: 960px) {
    grid-template-columns: 1fr 4fr 2fr;
    grid-template-areas:
      'header search nav'
      'content content content'
      'footer footer footer';

    nav {
      flex-direction: row;
    }
  }
`

const StyledLogo = styled(Logo)`
  grid-area: header;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  height: 4rem;
  & .logo_full {
    display: none;
    font-family: 'Monoton', cursive;
  }

  @media (min-width: 560px) {
    & .logo_short {
      display: none;
    }

    & .logo_full {
      display: inline;
    }
  }
`

const MainNav = styled.nav`
  grid-area: nav;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 0 2vmin;
  a {
    cursor: pointer;
    color: ${({ theme }) => theme.font.regular};
    &:hover {
      opacity: 0.7;
    }
  }
`

const LogoLink = styled(StyledLink)`
  padding-right: 1vw;
`

const SearchInput = styled(Input)`
  grid-area: search;
  width: 100%;
  height: 4rem;
`

const Content = styled.main`
  grid-area: content;
`

const Footer = styled.footer`
  grid-area: footer;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  height: 5rem;
`

function Layout({
  children,
  isDark,
  onThemeToggle,
}: {
  children: ReactNode
  isDark: boolean
  onThemeToggle: () => void
}) {
  return (
    <Wrapper>
      <Link href="/" passHref>
        <LogoLink>
          <StyledLogo size={3}>
            <span className="logo_short">C8X</span>
            <span className="logo_full">CoursesBox</span>
          </StyledLogo>
        </LogoLink>
      </Link>
      <MainNav>
        <Link href="/all">All</Link>
        <Link href="/news">News</Link>
        <IconButton
          name={isDark ? 'Sun' : 'Moon'}
          size={1}
          onClick={onThemeToggle}
        />
      </MainNav>
      <SearchInput icon="Search" placeholder="Search" onChange={() => {}} />
      <Content>{children}</Content>
      <Footer>
        Â© {new Date().getFullYear()} Ericsen Tsai. All rights reserved.
      </Footer>
    </Wrapper>
  )
}

export default Layout
