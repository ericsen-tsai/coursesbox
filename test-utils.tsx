import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
