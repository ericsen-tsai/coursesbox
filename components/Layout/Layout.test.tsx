import { render } from '@/test-utils'
import Layout from './index'

describe('Input test cases', () => {
  const child = (
    <>
      <h1>Main article area</h1>
      <p>
        In this layout, we display the areas in source order for any screen less
        then 500 pixels wide. We go to ta two column layout, and then to a three
        column layout by redefining the grid, and the placement of items on the
        grid.
      </p>
    </>
  )

  it('Render check', () => {
    const { asFragment } = render(<Layout>{child}</Layout>)

    expect(asFragment()).toMatchSnapshot()
  })
})
