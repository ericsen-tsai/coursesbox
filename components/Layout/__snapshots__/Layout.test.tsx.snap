// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test cases Render check 1`] = `
<DocumentFragment>
  <div
    class="css-qgkazi"
  >
    <a
      class="css-xh9t3y"
      href="/"
    >
      <header
        class="css-xgaehu"
        size="3"
      >
        <span
          class="logo_short"
        >
          C8X
        </span>
        <span
          class="logo_full"
        >
          CoursesBox
        </span>
      </header>
    </a>
    <nav
      class="css-to126x"
    >
      <a
        href="/all"
      >
        All
      </a>
      <a
        href="/news"
      >
        News
      </a>
      <button
        class="css-vxwc4q"
        size="2"
        title="Sun"
      >
        <svg
          aria-label="Sun"
          class="css-1tx9j83"
          fill="none"
          height="1rem"
          role="img"
          viewBox="0 0 24 24"
          width="1rem"
        >
          <path
            clip-rule="evenodd"
            d="M12 16a4 4 0 100-8 4 4 0 000 8zm0 2a6 6 0 100-12 6 6 0 000 12zM11 0h2v4.062a8.079 8.079 0 00-2 0V0zM7.094 5.68L4.222 2.808 2.808 4.222 5.68 7.094A8.048 8.048 0 017.094 5.68zM4.062 11H0v2h4.062a8.079 8.079 0 010-2zm1.618 5.906l-2.872 2.872 1.414 1.414 2.872-2.872a8.048 8.048 0 01-1.414-1.414zM11 19.938V24h2v-4.062a8.069 8.069 0 01-2 0zm5.906-1.618l2.872 2.872 1.414-1.414-2.872-2.872a8.048 8.048 0 01-1.414 1.414zM19.938 13H24v-2h-4.062a8.069 8.069 0 010 2zM18.32 7.094l2.872-2.872-1.414-1.414-2.872 2.872c.528.41 1.003.886 1.414 1.414z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </nav>
    <label
      class="css-a4q0le"
      height="7"
      width="20"
    >
      <div
        class="css-1v67xy3"
      >
        <input
          class="css-11hiddo"
          placeholder="Search"
        />
        <svg
          aria-hidden="true"
          aria-label="Search"
          class="css-gj6q3"
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="2rem"
          role="img"
          viewBox="64 64 896 896"
          width="2rem"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </div>
    </label>
    <main
      class="css-1qc1fvq"
    >
      <h1>
        Main article area
      </h1>
      <p>
        In this layout, we display the areas in source order for any screen less then 500 pixels wide. We go to ta two column layout, and then to a three column layout by redefining the grid, and the placement of items on the grid.
      </p>
    </main>
    <footer
      class="css-1cpmbek"
    >
      Â© 2022 Ericsen Tsai. All rights reserved.
    </footer>
  </div>
</DocumentFragment>
`;
